src/mouse-joystick-source.js

```diff
@@ -0,0 +1,179 @@
+
+        // Also listen to global mouse moves to track position even when not over canvas
+        document.addEventListener("mousemove", this.handleGlobalMouseMove);
+    }
+
+    /**
+     * Handle mouse movement over the canvas
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleMouseMove(event) {
+        if (!this.isActive) return;
+
+        const rect = this.canvas.getBoundingClientRect();
+        const x = event.clientX - rect.left;
+        const y = event.clientY - rect.top;
+
+        // Normalize to 0-1 range
+        this.mouseX = x / rect.width;
+        this.mouseY = y / rect.height;
+
+        // Clamp values
+        this.mouseX = Math.max(0, Math.min(1, this.mouseX));
+        this.mouseY = Math.max(0, Math.min(1, this.mouseY));
+    }
+
+    /**
+     * Handle mouse entering the canvas
+     */
+    handleMouseEnter() {
+        this.isActive = true;
+    }
+
+    /**
+     * Handle mouse leaving the canvas
+     */
+    handleMouseLeave() {
+        this.isActive = false;
+        // Don't center when mouse leaves - keep last position
+    }
+
+    /**
+     * Handle global mouse movement (even when not over canvas)
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleGlobalMouseMove(event) {
+        const rect = this.canvas.getBoundingClientRect();
+        const x = event.clientX - rect.left;
+        const y = event.clientY - rect.top;
+
+        // Normalize to 0-1 range
+        this.mouseX = x / rect.width;
+        this.mouseY = y / rect.height;
+
+        // Clamp values to 0-1 range
+        this.mouseX = Math.max(0, Math.min(1, this.mouseX));
+        this.mouseY = Math.max(0, Math.min(1, this.mouseY));
+    }
+
+    /**
+     * Handle mouse button press
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleMouseDown(event) {
+        if (!this.isActive || !this.via) return;
+
+        // Only handle left mouse button (button 0)
+        if (event.button === 0) {
+            // Set fire button 1 pressed (PB4)
+            this.via.setJoystickButton(0, true);
+            event.preventDefault();
+        }
+    }
+
+    /**
+     * Handle mouse button release
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleMouseUp(event) {
+        if (!this.via) return;
+
+        // Only handle left mouse button (button 0)
+        if (event.button === 0) {
+            // Release fire button 1 (PB4)
+            this.via.setJoystickButton(0, false);
+            event.preventDefault();
+        }
+    }
+
+    /**
+     * Set the VIA reference for button handling
+     * @param {object} via - The system VIA
+     */
+    setVia(via) {
+        this.via = via;
+    }
+
+    /**
+     * Get analog value from mouse position for the specified channel
+     * @param {number} channel - The ADC channel (0-3)
+     * @returns {number} A value between 0 and 0xffff
+     */
+    getValue(channel) {
+        let value;
```

Comment:
Please return directly instead of assigning to this variable and returning it separately. Then remove the extra braces on line 140. This function should be super simple.